library/src/main/java/org/apache/apex/malhar/lib/state/managed/Bucket.java

```diff
@@ -361,6 +405,16 @@ public Slice get(Slice key, long timeBucket, ReadSource readSource)
      */
     private BucketedValue getValueFromTimeBucketReader(Slice key, long timeBucket)
     {
+      if (bloomFilter != null) {
```

Comment:
How about the `getFromReaders` call then? That can be a place for using the bloom filter. Also once the value is fetched, it should immediately be inserted into the bloom filter. I don't understand how that would decrease the accuracy of the filter. Can you clarify the second part of your comment?